File: cloud-connector/src/main/java/no/sintef/cloudml/CloudConnectorFactory.java
package no.sintef.cloudml;

import no.sintef.cloudml.cloudconnector.JcloudsConnector;

public class CloudConnectorFactory {

	private CloudConnectorFactory() {
	}

	public static CloudConnectorFactory getInstance() {
		return CloudConnectorFactoryHolder.INSTANCE;
	}

	public static CloudConnector getCloudConnector() {
		return new JcloudsConnector();
	}

	private static class CloudConnectorFactoryHolder {

		private static final CloudConnectorFactory INSTANCE = new CloudConnectorFactory();
	}
}

File: cloud-connector/src/main/scala/no/sintef/cloudml/CloudConnectorFactory.scala
package no.sintef.cloudml;

import no.sintef.cloudml.cloudconnector.JcloudsConnector
import no.sintef.cloudml.connector.CloudConnector

class CloudConnectorFactory private (val instance: CloudConnector) { 
}

object CloudConnectorFactory {
    def create() = {
        new JcloudsConnector()
    }
}